### ISSUE TITLE
Bug: indent rules do not seem to apply to assignments as expected

### ISSUE THREAD

**Issue by bbuck** [eslint/eslint#17648]
### Environment

Node version: 18.16
npm version: pnpm 8.9.0
Local ESLint version: 8.48
Global ESLint version: N/A
Operating System: macOS

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "env": {
    "node": true
  }
  "rules": {
    "indent": [
      "error",
      2,
      {
        "VariableDeclarator": 2,
        "MemberExpression": 2,
        "FunctionDeclaration": {
          "parameters": 2
        },
        "FunctionExpression": {
          "parameters": 2
        },
        "CallExpression": {
          "arguments": 2
        },
        "offsetTernaryExpressions": true
      }
    ]
  }
}
```
</details>

```js
export const loadUserFromAuthorization =
  (_authorization: string): UserModel | null => {
    return null;
  };
```

### What did you expect to happen?

I expected the the assignment to fail lint pass because it's indented only one level, but the config for `VariableDeclarator` is defined to be two levels. Two levels of indentation at 2 spaces per level should be 4 spaces

```js
// expected passing result
export const loadUserFromAuthorization =
    (_authorization: string): UserModel | null => {
      return null;
    };
```

### What actually happened?

No lint error is flagged on the indentation levels of the assignment. There are no rules that define behavior specifically for assignment so I would assume that it falls back to this specific configuration.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgbG9hZFVzZXJGcm9tQXV0aG9yaXphdGlvbiA9XG4gIChfYXV0aG9yaXphdGlvbikgPT4ge1xuICAgIHJldHVybiBudWxsO1xuICB9OyIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnsiaW5kZW50IjpbImVycm9yIiwyLHsiVmFyaWFibGVEZWNsYXJhdG9yIjoyLCJNZW1iZXJFeHByZXNzaW9uIjoyLCJGdW5jdGlvbkRlY2xhcmF0aW9uIjp7InBhcmFtZXRlcnMiOjJ9LCJGdW5jdGlvbkV4cHJlc3Npb24iOnsicGFyYW1ldGVycyI6Mn0sIkNhbGxFeHByZXNzaW9uIjp7ImFyZ3VtZW50cyI6Mn0sIm9mZnNldFRlcm5hcnlFeHByZXNzaW9ucyI6dHJ1ZSwiU3dpdGNoQ2FzZSI6MCwiZmxhdFRlcm5hcnlFeHByZXNzaW9ucyI6ZmFsc2UsImlnbm9yZUNvbW1lbnRzIjpmYWxzZX1dfSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @bbuck, thanks for the issue. We aren't making further changes to the `indent` rule at this point. We recommend using a formatter like `prettier` or `dprint`.

**Comment by bbuck** [comment]
So this is unsupported behavior in `indent`? Your response suggests as much but I filed this a bug not a feature request as I would expect this behavior from the `indent` rule.

**Comment by Rec0iL99** [comment]
We are moving to deprecate formatting rules from ESLint core. Please track #17522 for more info. Thanks.
